apps/web/src/pages/quick-start/components/SetupTimeline.tsx

```diff
@@ -0,0 +1,133 @@
+  onConfigureLater?: () => void;
+}) => {
+  const { environment } = useEnvController();
+  const { data: apiKeys } = useQuery<{ key: string }[]>([QueryKeys.getApiKeys], getApiKeys);
+  const apiKey = apiKeys?.length ? apiKeys[0].key : '';
+  const { colorScheme } = useMantineColorScheme();
+  const isDark = colorScheme === 'dark';
+
+  const { data: inAppData } = useQuery<IGetInAppActivatedResponse>(
+    [QueryKeys.getInAppActive],
+    async () => getInAppActivated(),
+    {
+      refetchInterval: (data) => (data?.active ? false : 3000),
+      initialData: { active: false },
+    }
+  );
+
+  const instructions = frameworkInstructions.find((instruction) => instruction.key === framework)?.value ?? [];
+  const environmentIdentifier = environment?.identifier ? environment.identifier : '';
+
+  return (
+    <Stack align="center" sx={{ width: '100%' }}>
+      <TimelineWrapper isDark={isDark}>
+        <Timeline
+          active={instructions?.length + 1}
+          bulletSize={40}
+          lineWidth={2}
+          styles={{
+            itemBullet: {
+              backgroundColor: 'grey',
+            },
+          }}
+        >
+          {instructions.map((instruction, index) => {
+            return (
+              <Timeline.Item
+                bullet={
+                  <div style={{}}>
+                    <Text>{index + 1}</Text>
+                  </div>
+                }
+                key={index}
+                title={<div>{instruction.instruction}</div>}
+              >
+                <div style={{ marginTop: 10 }}>
+                  <PrismOnCopy
+                    language={instruction.language}
+                    index={index}
+                    code={`${updateCodeSnippet(instruction.snippet, environmentIdentifier, apiKey)}   `}
+                    onCopy={onCopy}
+                  />
+                </div>
+              </Timeline.Item>
+            );
+          })}
+          <Timeline.Item
+            bullet={
+              <div style={{}}>
+                <Text>{instructions?.length + 1}</Text>
+              </div>
+            }
+            title={'Render the components and run application'}
+          >
+            <LoaderWrapper>
+              <SetupStatus appInitialized={inAppData.active} onDone={onDone} onConfigureLater={onConfigureLater} />
+            </LoaderWrapper>
+          </Timeline.Item>
+        </Timeline>
+      </TimelineWrapper>
+
+      <When truthy={framework === 'demo'}>
+        <span style={{ color: colors.B60 }}>
+          If your browser did not automatically open, go to localhost at http://localhost:3000
+        </span>
+      </When>
+    </Stack>
+  );
+};
+
+const LoaderWrapper = styled.div`
+  margin-bottom: 20px;
+  margin-top: 10px;
+`;
+
+function updateCodeSnippet(codeSnippet: string, environmentIdentifier: string, apiKey: string) {
+  const concatUrls = process.env.REACT_APP_ENVIRONMENT !== 'production' || !!process.env.REACT_APP_DOCKER_HOSTED_ENV;
```

Comment:
```suggestion
  const concatUrls = ENV !== 'production' || !!IS_DOCKER_HOSTED;
```
And import them from the config in line 7.